src/WebJobs.Script/Binding/ExtensionBinding.cs

```diff
@@ -0,0 +1,255 @@
+                    }
+                }
+
+                // construct the attribute instance
+                var attributeConstructionInfo = GetAttributeConstructionInfo(attributeToCreate.Key, attributeData);
+                Attribute attribute = (Attribute)attributeConstructionInfo.Constructor.Invoke(attributeConstructionInfo.ConstructorArgs);
+
+                // apply any named property values
+                foreach (var namedProperty in attributeConstructionInfo.Properties)
+                {
+                    namedProperty.Key.SetValue(attribute, namedProperty.Value);
+                }
+
+                attributes.Add(attribute);
+            }
+
+            return attributes;
+        }
+
+        // TEMP
+        private string ResolveAndBind(string value, IReadOnlyDictionary<string, string> bindingData)
+        {
+            BindingTemplate template = BindingTemplate.FromString(value);
+
+            string boundValue = value;
+
+            if (bindingData != null)
+            {
+                if (template != null)
+                {
+                    boundValue = template.Bind(bindingData);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(value))
+            {
+                boundValue = Resolve(boundValue);
+            }
+
+            return boundValue;
+        }
+
+        internal class AttributeConstructionInfo
+        {
+            public ConstructorInfo Constructor { get; set; }
+            public object[] ConstructorArgs { get; set; }
+            public IDictionary<PropertyInfo, object> Properties { get; set; }
+        }
+
+        internal static CustomAttributeBuilder GetAttributeBuilder(Type attributeType, IDictionary<string, object> attributeData)
+        {
+            AttributeConstructionInfo constructionInfo = GetAttributeConstructionInfo(attributeType, attributeData);
+
+            var namedProperties = constructionInfo.Properties.Keys.ToArray();
+            var namedPropertyValues = constructionInfo.Properties.Values.ToArray();
+            CustomAttributeBuilder builder = new CustomAttributeBuilder(constructionInfo.Constructor, constructionInfo.ConstructorArgs, namedProperties, namedPropertyValues);
+
+            return builder;
+        }
+
+        internal static IDictionary<string, object> GetAttributeData(Attribute attribute)
+        {
+            Dictionary<string, object> data = new Dictionary<string, object>();
+
+            foreach (var property in attribute.GetType().GetProperties())
+            {
+                object value = property.GetValue(attribute);
+                if (value != null)
+                {
+                    data.Add(property.Name, value);
+                }
+            }
+
+            return data;
+        }
+
+        internal static AttributeConstructionInfo GetAttributeConstructionInfo(Type attributeType, IDictionary<string, object> attributeData)
+        {
```

Comment:
Was going through this but remembered you talking about using some existing code for the attribute construction. Is this it? If so, it's something that's already pretty well tested in the SDK, correct?